src/node/internal/process.ts

```diff
@@ -20,13 +20,90 @@ export function nextTick(cb: Function, ...args: unknown[]) {
   });
 }
 
-// Note that there is no process-level environment in workers so the process.env
-// object is initially empty. This is different from Node.js where process.env
-// picks up values from the operating system environment. The configured bindings
-// for the worker are accessible from the env argument passed into the fetch
-// handler and have no impact here.
+// Decide if a value can round-trip to JSON without losing any information.
+function isJsonSerializable(value: any): boolean {
+  switch (typeof value) {
+    case "boolean":
+    case "number":
+    case "string":
+      return true;
 
-export const env = new Proxy(utilImpl.getEnvObject(), {
+    case "object": {
+      if (value === null) {
+        return true;
+      }
+
+      // We only consider objects to be serializable if they are plain objects or plain arrays.
+      // Any subclass is not serializable because serializing it would be lossy.
+      //
+      // (Practically speaking, we don't want, say, a KV namespace to show up in process.env as
+      // "{}" -- this would be weird.)
+      switch (Object.getPrototypeOf(value)) {
+        case Object.prototype:
+          for (let prop of Object.values(value)) {
+            if (!isJsonSerializable(prop)) return false;
```

Comment:
If the object has circular references this recursion will be a problem. You'll likely want to memoize and bail out

Reply:
Ugh.

At present there's no way to express a cyclic object as an env var, and actually I would like to back out support of cycles in JSRPC because it's a security issue (because people tend to forget about cycles when writing code exactly like this).

But I suppose it would be good to not make assumptions here.

Two options:
* Use a Set to track what we've already seen.
* Don't actually do a deep scan. I was actually considering this anyway: Maybe we just say, if the top-level value is a simple object or simple array, we go ahead and JSON-serialize it. This does mean that if there's any nested non-JSON types, they may stringify as `{}`. But that doesn't seem so bad, and we don't have any extant case where that happens anyway.

Which do you prefer?

Reply:
Wait never mind, if we don't detect cycles then JSON.stringify() will actually throw an exception which will break everything.

So I guess we have to use the first option.

Reply:
Yep, using the set and bailing out (returning false) if cycles are detected would be the best option here.

Reply:
Done.